clang/include/clang/Basic/AddressSpaces.h

```diff
@@ -86,6 +86,24 @@ inline bool isPtrSizeAddressSpace(LangAS AS) {
           AS == LangAS::ptr64);
 }
 
+/// Convert an OpenCL lang address space into a SYCL one.
```

Comment:
Why do you need it?
BTW, I think we should use OpenCL address spaces "as is" instead of doing some stuff with SYCL address spaces.

Reply:
Yeah, there was a significant effort as I recall to remove this, @bader ?

Reply:
Yeah, here it is https://github.com/intel/llvm/pull/1039 . The only problem is failing C++ for OpenCL test but it is because of bug in C++ for OpenCL (I debugged it and described it in comments for PR 1039).

Reply:
> BTW, I think we should use OpenCL address spaces "as is" instead of doing some stuff with SYCL address spaces.

I totally agree with you, I don't see any rational to have this distinction.

The problem is that this patch is not merge, so for the moment `opencl_global` != `sycl_global`. This means a pointer to `opencl_global` is different to a pointer `sycl_global`.